### PR TITLE
feat(ci): setup kafka topic and fulfill with test data

### PR DESCRIPTION
**Pull request by tabVersion**
Signed-off-by: tabVersion <tabvision@bupt.icu>

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

as title

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]
Looks great to me!

**Review by skyzh — APPROVED** [review]

**Review by shanicky — APPROVED** [review]
LGTM

**Comment by neverchanje** [comment]
You can also add some e2e tests to verify the pumped data is available to read:

```
create materialized source s (v1 int, v2 varchar) with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_1_partition_topic', 'kafka.bootstrap.servers' = 'localhost:29092' ) row format 'json'

query IT rowsort
select * from s
----
1 1
2 22
3 333
4 4444

statement ok
drop source s
```

**Comment by tabVersion** [comment]
> You can also add some e2e tests to verify the pumped data is available to read:
> 
> ```
> create materialized source s (v1 int, v2 varchar) with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_1_partition_topic', 'kafka.brokers' = 'localhost:29092' ) row format json
> 
> query IT rowsort
> select * from s
> ----
> 1 1
> 2 22
> 3 333
> 4 4444
> 
> statement ok
> drop source s
> ```

we already have the basic tests, let's reopen them.

**Comment by neverchanje** [comment]
Yes, pls note that the sql syntax had changed a lot. The original tests can not be directly used.

File: e2e_test/source/basic_test.slt [hunk 1]
```diff
@@ -13,7 +13,7 @@ statement ok
 drop source s
 
 statement ok
-create source s (v1 int not null, v2 char(8) not null) with ( 'upstream.source' = 'kafka', 'kafka.topic' = 'kafka_2_partition_topic', 'kafka.bootstrap.servers' = 'localhost:29092' ) row format 'json'
+create materialized source s (v1 int not null, v2 varchar not null) with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_2_partition_topic', 'kafka.brokers' = '127.0.0.1:29092' ) row format json
```
Comment (neverchanje) [inline e2e_test/source/basic_test.slt:16]:
```suggestion
create materialized source s (v1 int, v2 varchar) with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_2_partition_topic', 'kafka.brokers' = '127.0.0.1:29092' ) row format json
```

It's no use to specify `not null`, we always ignore it internally.

Comment (tabVersion) [reply e2e_test/source/basic_test.slt]: fixed

**Comment by tabVersion** [comment]
close basic tests because the source reader is async and it is not created until `select`. It is possible to give either 4 rows or 0 rows as result.

In the current design, we cannot make it deterministic and sqllogictest cannot handle the situation. So I will close these tests and reopen them later.

**Pull request merged**